src/data_gradients/feature_extractors/object_detection/bounding_boxes_iou.py

```diff
@@ -61,7 +61,13 @@ def _compute_cumulative_counts_at_thresholds(self, df: pd.DataFrame, class_names
         return counts
 
     def aggregate(self) -> Feature:
-        df = pd.DataFrame(self.data).sort_values(by="class_id")
+        df = pd.DataFrame(self.data)
+
+        if len(df) == 0:
+            self._show_plot = False
```

Comment:
1) I don't think it is a good design to change the state of feature extractor to control whether the plot should happen or not. 
2) If the data is missing, I think the plotting logic is repponsible to deciding whether it can or cannot visualize the data.
3) This implementation has the bug IMO - self._show_plot is never reset to True afterwards. Maybe it is not affecting anything for now, but we may be planting a time bomb for future.

Reply:
> I don't think it is a good design to change the state of feature extractor to control whether the plot should happen or not.
> If the data is missing, I think the plotting logic is repponsible to deciding whether it can or cannot visualize the data.

Good point

> This implementation has the bug IMO - self._show_plot is never reset to True afterwards. Maybe it is not affecting anything for now, but we may be planting a time bomb for future.

The features are called only once and I don't see a reason for it to change in the future, so I don't think it's a big threat.  But yeah better safe than sorry

Reply:
@Louis-Dupont I suggest going back to your original design (before my lousy advice) and having the `aggregate` function return an object containing the description,  title, notice, error, and any other required flags. 
this way, everything will be much simpler... 
WDYT?